### ISSUE TITLE
[FR] Migrate to package:web from dart:html to support WASM-compilation

### ISSUE THREAD

**Issue by dxvid-pts** [AppFlowy-IO/appflowy-editor#811]
### Description

`dart:html` is superceeded by `package:web`. See https://dart.dev/interop/js-interop/package-web, https://docs.flutter.dev/platform-integration/web/wasm and this talk (https://www.youtube.com/watch?v=qx42r29HhcM) for more information.

### Impact

Projects targetting the web platform can benefit from using the suggested feature as they are now unblocked from compiling to WASM.

**Comment by mo-edumo** [comment]
I've opened #1223 with a fix for this: the only `dart:html` dependency in the library is `universal_html` in `platform_extension.dart` (reading `navigator.platform` for the `isWebOn*` getters), replaced there with `package:web` behind a `dart.library.js_interop` conditional import.

Verified in a consumer app: `flutter build web --wasm` compiles with the patched package, and the editor works at runtime on the dart2wasm build (typing, selection, Cmd+B shortcut on macOS web).

For anyone tracking: #1019, #1131 and #1141 are duplicates of this issue.
